### ISSUE TITLE
 Invalid CSRF Token. Unable to validate CSRF Token. 

### ISSUE THREAD

**Issue by chiragsatapara** [kunalvarma05/dropbox-php-sdk#85]
Invalid CSRF Token. Unable to validate CSRF Token.
File: D:\xampp\htdocs\arcade\api\libs\drop\vendor\kunalvarma05\dropbox-php-sdk\src\Dropbox\Authentication\DropboxAuthHelper.php

I checkout the `DropboxAuthHelper.php` file and found out that `$tokenInStore = $this->getPersistentDataStore()->get('state');` this line return a null value. Any change or something you want to suggest for this.

I am following this document : https://github.com/kunalvarma05/dropbox-php-sdk/wiki/Authentication-and-Authorization

**Comment by kunalvarma05** [comment]
Has the session been started? Make sure you've the `session_start();` on top of the script. 

**[See this Example](https://github.com/kunalvarma05/dropbox-php-sdk/wiki/Authentication-and-Authorization#example)** on the same page.

**Comment by jkbennemann** [comment]
Got the same error.
`session_start()` exists on top of the script.

**Comment by OctavioSI** [comment]
I am facing the same problem described here, using Laravel 5.4 and passing an endpoint in my API (routes/api.php) as redirect_uri to dropbox. 

After digging some possible causes to this problem, I have identified that line 178 of /Dropbox/Authentication/DropboxAuthHelper.php is likely to be the issue: 
`$tokenInStore = $this->getPersistentDataStore()->get('state');`

As it seems, $tokenInStore relies in a variable stored in session, as @kunalvarma05 mentioned above the need to include `session_start()` on top of the script. However, Laravel's api routes do not use sessions (it also seems to be bad practice to do so) and I've tried to create a custom middleware in my api route so as to enable the use of sessions, such as:

`Route::group(['middleware'=>'sessions'], function () { //precisa de session
      Route::get('dropbox/generatetoken', 'DropboxController@generateAccessToken');
    });`

Additionally, I had to create a middleware in Http/Middleware/Kernel.php:

`    protected $middlewareGroups = [
        'web' => [
            ...
        ],
        'api' => [
            ...
        ],
        'sessions' => [
            \Illuminate\Session\Middleware\StartSession::class,
        ]
    ];`

I was able to access the general $_SESSION variable this way, but the error remains. Logging $tokenInStore also shows that the variable is empty. 

I was able, however, to get the access token string after commenting the following lines in the validateCSRFToken function of DropboxAuthHelper.php as even mentioned in issue #105 
`       //Unable to fetch CSRF Token
        if (!$tokenInStore || !$csrfToken) {
             throw new DropboxClientException("Invalid CSRF Token. Unable to validate CSRF Token.");
        }
        //CSRF Token Mismatch
        if ($tokenInStore !== $csrfToken) {
             throw new DropboxClientException("Invalid CSRF Token. CSRF Token Mismatch.");
        }`

Obviously, this approach appears to be incorrect as CSRF validation is not being made. I wonder if there are other Laravel users who could put together this authentication using api.php
